### PR TITLE
[kube-prometheus-stack] Import Node Exporter / Nodes grafana dashboard conditionally

### PR DESCRIPTION
**Pull request by amuraru**
#### What this PR does / why we need it:
This dashboard depends on nodeExporter metrics
so make this dashboard depend on that being enabled

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/kube-prometheus-stack/templates/grafana/dashboards-1.14/nodes.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@ Do not change in-place! In order to change this file first read following link:
 https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack/hack
 */ -}}
 {{- $kubeTargetVersion := default .Capabilities.KubeVersion.GitVersion .Values.kubeTargetVersionOverride }}
-{{- if and (or .Values.grafana.enabled .Values.grafana.forceDeployDashboards) (semverCompare ">=1.14.0-0" $kubeTargetVersion) (semverCompare "<9.9.9-9" $kubeTargetVersion) .Values.grafana.defaultDashboardsEnabled }}
+{{- if and (or .Values.grafana.enabled .Values.grafana.forceDeployDashboards) (semverCompare ">=1.14.0-0" $kubeTargetVersion) (semverCompare "<9.9.9-9" $kubeTargetVersion) .Values.grafana.defaultDashboardsEnabled .Values.nodeExporter.enabled }}
```
Comment (monotek) [inline charts/kube-prometheus-stack/templates/grafana/dashboards-1.14/nodes.yaml:7]: https://github.com/prometheus-community/helm-charts/blob/b387aae0bb587bc6a97a35576d165b2d6bab8b0d/charts/kube-prometheus-stack/templates/grafana/dashboards-1.14/nodes.yaml#L3

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/kube-prometheus-stack/templates/grafana/dashboards-1.14/nodes.yaml (hunk 1 above)
Comment (amuraru) [reply charts/kube-prometheus-stack/templates/grafana/dashboards-1.14/nodes.yaml:7]: @monotek - sorry but I dont get your comment? Would u expand?

Comment (amuraru) [reply charts/kube-prometheus-stack/templates/grafana/dashboards-1.14/nodes.yaml:7]: @monotek that's what I did, see the change in charts/kube-prometheus-stack/hack/sync_grafana_dashboards.py and rerun to generate this file

**Comment by monotek** [comment]
Please rebase

**Review by monotek — APPROVED** [review]

File: charts/kube-prometheus-stack/Chart.yaml [hunk 2]
```diff
@@ -23,7 +23,7 @@ name: kube-prometheus-stack
 sources:
   - https://github.com/prometheus-community/helm-charts
   - https://github.com/prometheus-operator/kube-prometheus
-version: 32.0.2
+version: 32.0.3
```
Comment (monotek) [inline charts/kube-prometheus-stack/Chart.yaml:26]: new feature should raise minor version.

**Review by monotek — CHANGES_REQUESTED** [review]

**Review by monotek — APPROVED** [review]

**Pull request merged**